QandA/QandA.Rmd

```diff
@@ -599,16 +593,68 @@ knitr::include_graphics("../images/vectors.png")
 ## 4.1 Introduction
 
 :::question
-"There are three subsetting operators `[`. `[[`, `$`. What is the destinction between an operator and a function? When you look up the help page it brings up the same page for all three extraction methods. What are their destinctions and do their definitions change based on what you're subsetting? Can we make a table? 
+"There are three subsetting operators `[`. `[[`, `$`. What is the distinction between an operator and a function? When you look up the help page it brings up the same page for all three extraction methods. What are their distinctions and do their definitions change based on what you're subsetting? Can we make a table? 
+:::
+
+:::TODO
+```{r, echo=FALSE}
+knitr::include_graphics("../images/subsetting.png")
+```
+:::
+
+## 4.2.1 Selecting multiple elements
+
+:::question
+Why is `numeric(0)` "helpful for test data?"
+:::
+
+This is more of a general comment that one should make sure one's code doesn't crash with vectors of zero length (or data frames with zero rows)
+
+:::question
+Why is subsetting with factors "not a good idea"
+:::
+
+Hadley's notes seem to say subsetting with factors uses the "integer vector of levels" - and if they all have the same level, it'll just return the first argument. Subsetting a factor vector leaves the factor levels behind unless you explicitly drop the unused levels
+
+## 4.2.2 lists
+
+:::question
+We've been talking about `$` as a shorthand for `[[`. Using the example list `x <- list(1:3, "a", 4:6)` can we use `x$1` as shorthand for `x[[1]]`?
+:::
+
+The "shorthand" refers to using the name of the vector to extract the vector. If we give `1:3` a name such as test = `1:3`
+
+```{r}
+x <- list(named_vector = 1:3, "a", 4:6)
+x[[1]] == x$named_vector
+```
+
+As such, `$` is a shorthand for `x[["name_of_vector"]]` and not shorthand for `x[[index]]`
+
+## 4.3.1 `[[`
+
+:::question
+"While you must use [[ when working with lists, I’d also recommend using it with atomic vectors whenever you want to extract a single value. For example, instead of writing:"
+
+```{r, eval=FALSE}
+for (i in 2:length(x)) {
```

Comment:
Should `out[1]` be defined here?

Reply:
This question still hasn't been answered in the channel, this is an excerpt from the chapter that I want to make concrete with actual values for `x`, `out`, and `fun`.

Reply:
Think we can merge this then work on the answer in a new PR, @russHyde ? I'd like to add this since I did one mega PR with updates to the README (i'll make that separate in the future)

Reply:
I was (at least a tiny bit) driven to answer the question because it was flagged as unresolved here 🙃